test/jdk/java/util/HexFormat/HexFormatTest.java

```diff
@@ -0,0 +1,464 @@
+                Assert.assertTrue(value.startsWith(prefix), "wrong prefix");
+                Assert.assertTrue(value.endsWith(suffix), "wrong suffix");
+
+                // Check case of digits
+                String cc = value.substring(prefix.length(), prefix.length() + 2);
+                Assert.assertEquals(cc,
+                        (uppercase) ? cc.toUpperCase(Locale.ROOT) : cc.toLowerCase(Locale.ROOT),
+                        "Case mismatch");
+                if (i < expected.length - 1 && !delimiter.isEmpty()) {
+                    // Check the delimiter is present for each pair except the last
+                    Assert.assertEquals(res.substring(offset + valueChars,
+                            offset + valueChars + delimiter.length()), delimiter);
+                }
+            }
+        }
+    }
+
+    @Test(dataProvider="HexFormattersParsers")
+    static void testDecoderString(String unused1, String unused2, String unused3,
+                                   boolean unused4, HexFormat hex) {
+        byte[] expected = genBytes('A', 15);
+        String s = hex.formatHex(expected);
+        System.out.println("    formatted: " + s);
+
+        byte[] actual = hex.parseHex(s);
+        System.out.println("    parsed as: " + Arrays.toString(expected));
+        int mismatch = Arrays.mismatch(expected, actual);
+        Assert.assertEquals(actual, expected, "encode/decode cycle failed, mismatch: " + mismatch);
+    }
+
+    @Test(dataProvider="HexFormattersParsers")
+    static void testDecoderCharArray(String unused1, String unused2, String unused3,
+                                     boolean unused4, HexFormat hex) {
+        byte[] expected = genBytes('A', 15);
+        String s = hex.formatHex(expected);
+        System.out.println("    formatted: " + s);
+
+        char[] chars = s.toCharArray();
+        byte[] actual = hex.parseHex(chars, 0, chars.length);
+        System.out.println("    parsed as: " + Arrays.toString(expected));
+        int mismatch = Arrays.mismatch(expected, actual);
+        Assert.assertEquals(actual, expected, "format/parse cycle failed, mismatch: " + mismatch);
+    }
+
+    @Test(dataProvider="HexFormattersParsers")
+    static void testFormatterToString(String delimiter, String prefix, String suffix,
+                                    boolean uppercase,
+                                    HexFormat hex) {
+        String actual = String.format(
+                "uppercase: %s, delimiter: \"%s\", prefix: \"%s\", suffix: \"%s\"",
+                uppercase, escapeNL(delimiter), escapeNL(prefix), escapeNL(suffix));
+        System.out.println("    hex: " + actual);
+        Assert.assertEquals(actual, hex.toString(), "Formatter toString mismatch");
+    }
+
+    private static String escapeNL(String string) {
+        return string.replace("\n", "\\n")
+                .replace("\r", "\\r");
+    }
+
+    @Test
+    static void testVariableLength() {
```

Comment:
This test does not contain any assertions.

Reply:
Added missing assertions.